test/engine-luatest/gh_10847_memtx_sort_data_test.lua

```diff
@@ -0,0 +1,427 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g_generic = t.group('memtx_sort_data_test-generic')
+local g_sk_types = t.group('memtx_sort_data_test-sk-types', t.helpers.matrix {
+    pk_type = {'TREE', 'HASH'},
+    sk_part_type = {'unsigned', 'string'},
+    sk_part_sort_order = {'asc', 'desc'},
+    sk_extra = {
+        {hint = true},
+        {hint = false},
+        {path = 'scalar'},
+        {path = 'array[*]'}
+    }
+})
+
+g_sk_types.before_all(function(cg)
+    cg.server = server:new({
+        box_cfg = {memtx_sort_data_enabled = true},
+        alias = 'master'
+    })
+    cg.server:start()
+end)
+
+g_sk_types.after_all(function(cg)
+    cg.server:stop()
+end)
+
+g_sk_types.after_each(function(cg)
+    cg.server:exec(function()
+        if box.space.test then
+            box.space.test:drop()
+        end
+    end)
+end)
+
+-- Test that the SK sort using the sort data is correct.
+g_sk_types.test_memtx_sort_data = function(cg)
+    -- Create the space and indexes and write snapshot with sort data.
+    cg.server:exec(function(pk_type, sk_part_type,
+                            sk_part_sort_order, sk_extra)
+        local s = box.schema.create_space('test')
+        s:create_index('pk', {type = pk_type, parts = {1, 'unsigned'}})
+        s:create_index('sk', {
+            type = 'tree', hint = sk_extra.hint, unique = false,
+            parts = {2, sk_part_type, sort_order = sk_part_sort_order,
+                     path = sk_extra.path},
+        })
+
+        local function sk_object()
+            local function sk_value()
+                local max = 1000000
+                if sk_part_type == 'unsigned' then
+                    return math.random(1, max)
+                elseif sk_part_type == 'string' then
+                    return tostring(math.random(1, max))
+                else
+                    t.assert(false) -- can't insert this type
+                end
+            end
+
+            if sk_extra.path ~= nil then
+                local count = math.random(1, 3)
+                local array = {}
+                for i = 1, count do
+                    table.insert(array, sk_value())
+                end
+                return {array = array, scalar = sk_value()}
+            else
+                return sk_value()
+            end
+        end
+
+        for i = 1, 1000 do
+            s:insert({i, sk_object()})
+        end
+
+        box.snapshot()
+    end, {cg.params.pk_type, cg.params.sk_part_type,
+          cg.params.sk_part_sort_order, cg.params.sk_extra})
+
+    -- Start using the sort data.
+    cg.server:restart()
```

Comment:
Need to check the sort data file is used and was not ignored here (it's the case for `HASH` since it does not fill the old2new map now).

Reply:
Done.